plugins/fluent-bit/src/plugin.cpp

```diff
@@ -0,0 +1,153 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/chunk.hpp>
+#include <tenzir/data.hpp>
+#include <tenzir/error.hpp>
+#include <tenzir/logger.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <fluent-bit/fluent-bit-minimal.h>
+
+using namespace std::chrono_literals;
+
+namespace tenzir::plugins::fluentbit {
+
+namespace {
+
+struct operator_args {
+  located<std::vector<std::string>> args;
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, operator_args& x) -> bool {
+    return f.object(x)
+      .pretty_name("operator_args")
+      .fields(f.field("args", x.args));
+  }
+};
+
+class fluent_bit_operator final : public crtp_operator<fluent_bit_operator> {
+public:
+  fluent_bit_operator() = default;
+
+  explicit fluent_bit_operator(operator_args args) : args_{std::move(args)} {
+  }
+
+  auto operator()(operator_control_plane& ctrl) const
+    -> generator<table_slice> {
+    auto ctx = flb_create();
+    auto destroyer = caf::detail::make_scope_guard([ctx] {
+      flb_destroy(ctx);
+    });
+    if (ctx == nullptr)
+      diagnostic::error("failed to create fluent bit context")
+        .hint("flb_create returned nullptr")
+        .emit(ctrl.diagnostics());
+    // Set services properties.
```

Comment:
You probably want a `return` whenever an error is emitted.

Reply:
I sprinkled `co_return` statements across the call sequence.